src/Session.php

```diff
@@ -123,75 +105,32 @@ class Session implements \IteratorAggregate
 	/**
 	 * Constructor
 	 *
-	 * @param   string  $store    The type of storage for the session.
-	 * @param   array   $options  Optional parameters
+	 * @param   StorageInterface  $store    A StorageInterface implementation
+	 * @param   array             $options  Optional parameters
 	 *
 	 * @since   1.0
 	 */
-	public function __construct($store = 'none', array $options = array())
+	public function __construct(StorageInterface $store = null, array $options = array())
 	{
-		// Need to destroy any existing sessions started with session.auto_start
-		if (session_id())
-		{
-			session_unset();
-			session_destroy();
-		}
-
-		// Disable transparent sid support
-		ini_set('session.use_trans_sid', '0');
-
-		// Only allow the session ID to come from cookies and nothing else.
-		ini_set('session.use_only_cookies', '1');
-
-		// Create handler
-		$this->store = Storage::getInstance($store, $options);
-
-		$this->storeName = $store;
+		$this->store = $store ?: new NativeStorage(new FilesystemHandler);
 
 		// Set options
-		$this->_setOptions($options);
-
-		$this->_setCookieParams();
+		$this->setOptions($options);
+		$this->setCookieParams();
```

Comment:
There's no reason why you can't inject the options into Session directly into the driver. As the StorageInterface is a parameter of the Session object anyhow anyone who wants to can inject those options for cookies anyhow.

FWIW NativeStorage also takes a options array in the constructor so you could directly inject the session options in there anyhow